analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.CSharp9.cs

```diff
@@ -49,15 +49,16 @@ public void PatternMatching(object arg)
         }
     }
 
-    public void StaticLambda()
+    public void StaticLambda(object o)
     {
-        Func<object, string> a = static (arg) =>
-        {
-            return arg.ToString();      // Compliant, it's not a public method argument
-        };
-        a(null);
+        Func<object, string> lambda = static (object obj) => obj.ToString(); // Compliant - not accessible from other assemblies
+        lambda(o);
+
+        MethodAcceptsFunction(static obj => obj.ToString());
     }
 
+    private string MethodAcceptsFunction(Func<object, string> func) => func(null);
```

Comment:
For simplicity, what the body does is just a scaffolding noise that doesn't change the UT
```suggestion
    private void MethodAcceptsFunction(Func<object, string> func) { }
```